fix(docs-infra): update `getAnswerFiles` to ensure compatibility with non-POSIX file systems

alan-agius4 opened the pull request:
This update modifies the `getAnswerFiles` function to support file systems that do not adhere to POSIX standards.

Prior to this change this method will always fail with the below:

```
Error: Invalid state: could not find start of answers path
```

alan-agius4 commented on adev/shared-docs/pipeline/tutorials/metadata.ts:
```diff
@@ -49,18 +49,14 @@ async function getAnswerFiles(
-      const parentDir = dirname(answersDir) + '/';
-      const pathStart = absolutePath.indexOf(parentDir);
-      if (pathStart === -1) {
-        throw new Error('Invalid state: could not find start of answers path');
```
I am not quite understanding the reasoning behind this check, the `glob`, `cwd` is set to `answersDir`, thus returned file should always children of `answersDir`

devversion replied:
Maybe just a safety check that they did?

alan-agius4 replied:
IDK, I can leave it but it should never trigger unless there is a bug in the `glob` library.

devversion replied:
I think removing is good, with the approach you now have `absolute: false`

devversion reviewed (approved):
Much cleaner. LGTM

alan-agius4 committed:
fix(docs-infra): update `getAnswerFiles` to ensure compatibility with non-POSIX file systems

This update modifies the `getAnswerFiles` function to support file systems that do not adhere to POSIX standards.

Prior to this change this method will always fail with the below:

```
Error: Invalid state: could not find start of answers path
```

alxhub commented:
This PR was merged into the repository by commit 8f73199e9f03918bbb5d22f09306033de331f5c9.

The changes were merged into the following branches: main, 18.2.x
